pkg/piecestore/rpc/server/psdb/psdb.go

```diff
@@ -206,6 +238,31 @@ func (db *DB) GetBandwidthAllocationBySignature(signature []byte) ([][]byte, err
 	return agreements, nil
 }
 
+// GetBandwidthAllocations all bandwidth agreements and sorts by satellite
+func (db *DB) GetBandwidthAllocations() (map[string][]*Agreement, error) {
+	defer db.locked()()
+
+	rows, err := db.DB.Query(`SELECT * FROM bandwidth_agreements ORDER BY satellite`)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = rows.Close()
+	}()
+
+	agreements := make(map[string][]*Agreement)
+	for rows.Next() {
+		var agreement *Agreement
+		var satellite string
+		err := rows.Scan(&satellite, &agreement.Agreement, &agreement.Signature)
+		if err != nil {
+			return agreements, err
```

Comment:
Shouldn't `agreements` be `nil` to avoid returning partial results?

Reply:
changed to nil